fendermint/vm/actor_interface/src/ipc.rs

```diff
@@ -167,35 +175,55 @@ pub mod gateway {
         pub min_collateral: U256,
         pub msg_fee: U256,
         pub majority_percentage: u8,
+        pub validators: Vec<GenesisValidator>,
     }
 
-    impl TryFrom<GatewayParams> for ConstructorParameters {
-        type Error = fvm_shared::address::Error;
-
-        fn try_from(value: GatewayParams) -> Result<Self, Self::Error> {
+    impl ConstructorParameters {
+        pub fn new(
+            params: GatewayParams,
+            validators: Vec<Validator<Collateral>>,
+        ) -> Result<Self, AddressError> {
+            // Every step along the way in the subnet ID we have an Ethereum address.
             let mut route = Vec::new();
-            for addr in value.subnet_id.children() {
+            for addr in params.subnet_id.children() {
                 let addr = match addr.payload() {
                     Payload::ID(id) => EthAddress::from_id(*id),
                     Payload::Delegated(da)
                         if da.namespace() == eam::EAM_ACTOR_ID && da.subaddress().len() == 20 =>
                     {
                         EthAddress(da.subaddress().try_into().expect("checked length"))
                     }
-                    _ => return Err(fvm_shared::address::Error::InvalidPayload),
+                    _ => return Err(AddressError::InvalidPayload),
                 };
                 route.push(H160::from(addr.0))
             }
+
+            // Every validator has an Ethereum address.
+            let validators = validators
+                .into_iter()
+                .map(|v| {
+                    let pk = v.public_key.0.serialize();
```

Comment:
I will update the [PR](https://github.com/consensus-shipyard/ipc-solidity-actors/pull/228) so you can use the right type.